# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/debugger/Script.cpp
# Commit: e590eb34efc5
# Full Hash: e590eb34efc52ab109f2737167cb2aa0975a1792
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-06-01 09:43:14
# Regressor Bug: 1554524
# File Overlap Count: 1
# Description:
#   Bug 1624792: Remove Debugger interface r=tcampbell
#   
#   This patch removes the changes from bug 1554524 part 4. (Parts 1 and 3 of bug 1554524 were bytecode emitter cleanups that don't need to be reverted; part 2 was abandoned.)
#   
#   Depends on D116308
# ==============================================================================

diff -r 135a0e6af641 -r e590eb34efc5 js/src/debugger/Script.cpp
--- a/js/src/debugger/Script.cpp	Tue Jun 01 01:31:21 2021 +0000
+++ b/js/src/debugger/Script.cpp	Tue Jun 01 01:31:21 2021 +0000
@@ -246,7 +246,6 @@
   bool clearAllBreakpoints();
   bool isInCatchScope();
   bool getOffsetsCoverage();
-  bool setInstrumentationId();
 
   using Method = bool (CallData::*)();
 
@@ -2306,27 +2305,6 @@
   return true;
 }
 
-bool DebuggerScript::CallData::setInstrumentationId() {
-  if (!ensureScriptMaybeLazy()) {
-    return false;
-  }
-
-  if (!obj->getInstrumentationId().isUndefined()) {
-    JS_ReportErrorASCII(cx, "Script instrumentation ID is already set");
-    return false;
-  }
-
-  if (!args.get(0).isNumber()) {
-    JS_ReportErrorASCII(cx, "Script instrumentation ID must be a number");
-    return false;
-  }
-
-  obj->setReservedSlot(INSTRUMENTATION_ID_SLOT, args.get(0));
-
-  args.rval().setUndefined();
-  return true;
-}
-
 /* static */
 bool DebuggerScript::construct(JSContext* cx, unsigned argc, Value* vp) {
   JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr, JSMSG_NO_CONSTRUCTOR,
@@ -2366,7 +2344,6 @@
     JS_DEBUG_FN("getOffsetMetadata", getOffsetMetadata, 1),
     JS_DEBUG_FN("getOffsetsCoverage", getOffsetsCoverage, 0),
     JS_DEBUG_FN("getEffectfulOffsets", getEffectfulOffsets, 1),
-    JS_DEBUG_FN("setInstrumentationId", setInstrumentationId, 1),
 
     // The following APIs are deprecated due to their reliance on the
     // under-defined 'entrypoint' concept. Make use of getPossibleBreakpoints,